lib/questionnaires/model/src/questionnaire_item_model.dart

```diff
@@ -185,6 +185,44 @@ class QuestionnaireItemModel extends ChangeNotifier with Diagnosticable {
     }
   }
 
+  /// Returns an integer, starting with 1, that provides the number
+  /// of [QuestionnaireModel]s that have [isAnswerable] flags set to true
+  ///
+  int getQuestionNumber(int answerIndex) {
+    late final int questionNumber;
+
+    /// If [answerIndex] falls within the _cachedAnswerModels data set...
+    /// Check each question in turn until [answerIndex] is reached.
+    /// Create a count of all questions that are labeled as answerable until
+    /// [answerIndexx]
+    ///
+    if (_orderedItems != null) {
```

Comment:
Why do you not make sure it exists? There is a private _ensureOrderedItems function.

Reply:
This only runs if `_orderedItems` has been set from a nullable variable `LinkedHashMap<String, QuestionnaireItemModel>?` to a non-nullable one `LinkedHashMap<String, QuestionnaireItemModel>`.

Do you envision there will be a time when I need to run _ensureOrderedItems within this method? If so, I can make that change easily enough.